server/src/main/scala/de/frosner/broccoli/templates/TemplateConfig.scala

```diff
@@ -1,13 +1,55 @@
 package de.frosner.broccoli.templates
 
-import de.frosner.broccoli.models.{ParameterInfo, ParameterType}
+import com.typesafe.config.{ConfigObject, ConfigValue}
+import de.frosner.broccoli.models.{ParameterType, ParameterValue}
+import pureconfig.ConfigReader
+import pureconfig.error.{ConfigReaderFailures, ThrowableFailure}
+
+import scala.collection.JavaConversions._
+import scala.util.{Failure, Success, Try}
 
 object TemplateConfig {
 
+  implicit val configReader: ConfigReader[TemplateInfo] = new ConfigReader[TemplateInfo] {
```

Comment:
Wow why did you define your own config reader? Could it be better to keep the default behaviour from pure-config and parse the default value as `String` like it was. We then can parse the default value into the real type using the type information? Just thinking that this looks hard to maintain.

Reply:
IMO if we go the String way it kind of lands us into the same problem as before. Maybe @Gerrrr can comment.